buckets/redis/bucket.go

```diff
@@ -80,9 +80,9 @@ func (a *abstractBucket) Take(ctx context.Context, requested int64, maxWaitTime
 }
 
 func (a *abstractBucket) takeFromRedis(ctx context.Context, client *redis.Client, args []interface{}) *redis.Cmd {
-	span, ctx := opentracing.StartSpanFromContext(ctx, "EvalSha")
+	span, ctx := opentracing.StartSpanFromContext(ctx, "script.Run")
```

Comment:
Have you done any benchmarking with script.Run compared to executing a pre-loaded script?

Reply:
I haven't. `script.Run` should have the same amortized performance as `EvalSha`, or at least comparable performance.

Redis will [cache scripts](https://redis.io/commands/eval#script-cache-semantics) that are run with `EVAL`, so initial requests when redis comes up will fail, but then all subsequent `EVALSHA`'s will succeed until redis restarts/the script changes. It's a small penalty for the requests after a server restarts, but I think it should be _relatively_ minimal.

Reply:
`script.Run` calls `EVALSHA`, followed by `EVAL` if it fails ([implementation](https://github.com/go-redis/redis/blob/master/script.go#L56-L62)).

Reply:
Nice